src/account_manager/operations/participation.rs

```diff
@@ -62,6 +62,19 @@ impl AccountManager {
             .collect())
     }
 
+    pub async fn get_participation_events_from_client(
+        &self,
+        event_type: Option<ParticipationEventType>,
+    ) -> crate::Result<Vec<EventId>> {
+        let accounts = self.accounts.read().await;
+        Ok(if let Some(account) = accounts.first() {
+            let events = account.client.events(event_type).await?;
+            events.event_ids
+        } else {
+            vec![]
```

Comment:
If there is no client we should build it, similar to here
https://github.com/iotaledger/wallet.rs/blob/6b3cdfed7bf4960b2c682a4b10353df58711f294/src/account_manager/mod.rs#L165-L169